### PR TITLE
New master

### PR DESCRIPTION
**Pull request by ftalbrecht**
As the name suggests, this is to be the new master (as in force pushed over the current master), after the release. This makes #111 obsolete.

### CODE REVIEW

**Comment by renemilk** [comment]
Looks like you lost the CI changes I grafted onto the refactor branch yesterday.

File: dune/gdt/local/bilinear-forms/integrals.hh [hunk 1]
```diff
@@ -96,7 +96,9 @@ public:
     result *= 0;
     // loop over all quadrature points
     const size_t integrand_order = integrand_->order(test_basis, ansatz_basis) + over_integrate_;
-    for (const auto& quadrature_point : QuadratureRules<D, d>::rule(element.geometry().type(), integrand_order)) {
+    assert(integrand_order < std::numeric_limits<int>::max());
```
Comment (ftalbrecht) [inline dune/gdt/local/bilinear-forms/integrals.hh]: This should not be required and is a left-over from the old master. All `order()` methods now return ints, so the `integrand_order` is already an `int`. Thus we neither need the assertion, nor the `static_cast`. (As a side note: we do have the `numeric_cast` in dune-xt-common to perform the check and the cast, and to give an error message on failure.) Long story short, the following code should suffice:
```c++
const auto integrand_order = integrand_->order(test_basis, ansatz_basis) + over_integrate_;
for (const auto& quadrature_point : QuadratureRules<D, d>::rule(element.geometry().type(), integrand_order)) {
// ...
```

File: examples/stationary-heat-equation.cc [hunk 2]
```diff
@@ -79,15 +83,21 @@ int main(int argc, char* argv[])
...
     boundary_info.register_new_normal({0., -1.}, new XT::Grid::NeumannBoundary());
-    boundary_info.register_new_normal({0., 1.}, new XT::Grid::NeumannBoundary());
-    boundary_info.register_new_normal({-1., 0.}, new XT::Grid::DirichletBoundary());
+    boundary_info.register_new_normal({0., 1.}, new XT::Grid::DirichletBoundary());
+    boundary_info.register_new_normal({-1., 0.}, new XT::Grid::NeumannBoundary());
     boundary_info.register_new_normal({1., 0.}, new XT::Grid::DirichletBoundary());
 
+    // Only Dirichlet boundary conditions, in that case we need to change the domain to [-1,1]^2 to ensure correct
+    // boundary conditions
+    // auto grid = XT::Grid::make_cube_grid<G>(/*lower_left=*/-1., /*upper_right=*/1., /*num_elements=*/128);
```
Comment (ftalbrecht) [inline examples/stationary-heat-equation.cc]: Sorry for that, I also have related changes locally. What is true is that we need the `[-1, 1]^2` cube here for the exact solution and the boundary values to make sense. For the same reason, we should not treat Neumann boundary values here at all. So I would propose to only have these lines here:
```c++
auto grid = XT::Grid::make_cube_grid<G>(/*lower_left=*/-1., /*upper_right=*/1., /*num_elements=*/128);
auto grid_view = grid.leaf_view();

const XT::Grid::AllDirichletBoundaryInfo<I> boundary_info;
```

Comment (tobiasleibner) [reply examples/stationary-heat-equation.cc]: I think the the domain [0,1]^2 with Neumann boundary conditions on the left and bottom side also makes sense, as the gradient of the exact solution is zero on these boundaries. But I am happy with changing the file as you suggested.

**Comment by tobiasleibner** [comment]
Look as if the CI is still broken here...

**Comment by tobiasleibner** [comment]
Tests are compiling with both gcc and clang finally. There are still some failing tests, see [drone](https://drone.pymor.org/dune-community/dune-gdt/203/5). 

Failing tests:

	  2 - test_inviscid_compressible_flow__euler_1d__explicit__fv (Timeout)
	  3 - test_linear_transport__1d__explicit__dg_p0 (Timeout)
	  4 - test_linear_transport__1d__explicit__fv (Timeout)
	  5 - test_linear_transport__1d__implicit__fv (Failed)
	  9 - test_spaces_h1_continuous_lagrange__simplicial_grids (Failed)
	 13 - test_spaces_l2_discontinuous_galerkin__mixed_grids__scalar (Failed)
	 14 - test_spaces_l2_discontinuous_galerkin__mixed_grids__vector_valued (Failed)
         19 - test_spaces_l2_finite_volume (Failed)

**Comment by ftalbrecht** [comment]
> Tests are compiling with both gcc and clang finally. There are still some failing tests, see [drone](https://drone.pymor.org/dune-community/dune-gdt/203/5).
> 
> Failing tests:
> 
> ```
>   2 - test_inviscid_compressible_flow__euler_1d__explicit__fv (Timeout)
>   3 - test_linear_transport__1d__explicit__dg_p0 (Timeout)
>   4 - test_linear_transport__1d__explicit__fv (Timeout)

These timeouts are a bit straange, since I thought the tests run rather quick locally.

>   5 - test_linear_transport__1d__implicit__fv (Failed)

I can look into that, not too important.

>   9 - test_spaces_h1_continuous_lagrange__simplicial_grids (Failed)
>  13 - test_spaces_l2_discontinuous_galerkin__mixed_grids__scalar (Failed)
>  14 - test_spaces_l2_discontinuous_galerkin__mixed_grids__vector_valued (Failed)
>      19 - test_spaces_l2_finite_volume (Failed)

These failures are severe. From a quick glance it looks like the mapper has a problem...

**Comment by tobiasleibner** [comment]
Test are finally passing, except for a timeout in ``test_inviscid_compressible_flow__euler_1d__explicit__fv`` which seems to be a real timeout. The test takes 800 seconds in debug mode (release mode 40 seconds) on my laptop, so it is reasonable that it might take more than 1000s on drone. So we either need to improve the performance in debug mode or just shrink that test.

**Comment by ftalbrecht** [comment]
No objections. I have further changes locally bu these can just enter as prs at some point...

**Comment by tobiasleibner** [comment]
I saved the old ``master`` in the ``oldmaster`` branch and force-pushed this branch over the old ``master``.

**Pull request merged**